### PR TITLE
Add a reward example using ProLIF.

### PR DESCRIPTION
**Pull request by tyszw**
Added an example of a reward function that evaluates interactions between a receptor and generated molecules using ProLIF. Please confirm eb6e2c7263dfc3423da074608b8300a0eb4d2bed for details.

### CODE REVIEW

File: reward/util.py [hunk 1]
```diff
@@ -506,3 +507,81 @@ def calc_strain_energy(docking_pose_file, conf):
             if os.path.exists(file_path): os.remove(file_path)
 
     return total_strain_energy, max_single_strain_energy
+
+
+def get_interaction_distances(receptor_fname, output_ligand_fname, pose_idx, conf):
```
Comment (sishida21) [inline reward/util.py:512]: Add a comment to note that this function is currently support GNINA.

File: config/setting_gnina_interaction.yaml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+# Setting for GNINA
+gnina_bin_path: "reward/gnina.sif"
+gnina_receptor: /scr/5dxh_prep_rec2.pdb
+gnina_autobox_ligand: /scr/5h2_lig.mol2
+gnina_cpus: 4
+gnina_num_modes: 9
+gnina_base_smina_affinity: -10.01144
+gnina_base_cnn_affinity: 8.1709308624
+
+# Setting for prolif
```
Comment (sishida21) [inline config/setting_gnina_interaction.yaml:70]: Modify the key names related to ProLIF as follows to clarify that they are ProLIF options.

prolif_interactions: ...
prolif_tolerance: ...

File: reward/README.md [hunk 3]
```diff
@@ -129,3 +129,13 @@ Using this reward fucntion, you need to create Tldr account at [Tldr Web site](h
 You also need to install openbabel. Click [here](#here) for instllation instructions.
 
 Strainfilter is licensed by the [Irwin](https://irwinlab.compbio.ucsf.edu/) and [Shoichet](https://bkslab.org/) Laboratories in the Department of Pharmaceutical Chemistry at the University of California, San Francisco (UCSF).
+
+### gnina_interaction_reward.py
+
+This reward function assesses the interactions between the receptor and the generated molecules using [ProLIF](https://doi.org/10.1186/s13321-021-00548-6).
+Need to install prolif and MDAnalysis.
```
Comment (sishida21) [inline reward/README.md:136]: Add descriptions about how to specify the residue, interaction_type, cutoff, and etc.

e.g.,
residue: [residue_name][number].[chain]

**Comment by sishida21** [comment]
@tyszw Please address the comments.

**Review by sishida21 — COMMENTED** [review]
@tyszw Please address the comments.

File: config/setting_gnina_interaction.yaml (hunk 2 above)
Comment (tyszw) [reply config/setting_gnina_interaction.yaml:70]: Rename keys in e129ac4adeb1b5a670eac7fac4482c33fc761a38.

File: reward/README.md (hunk 3 above)
Comment (tyszw) [reply reward/README.md]: Added parameter descriptions in b91f91c4caa652403aac80f2482c458c021d5417.

**Comment by tyszw** [comment]
@sishida21 Thank you for your review. I have addressed your comments. Please let me know if there are any further adjustments needed.

**Pull request merged**